contracts/eden/src/migrations.cpp

```diff
@@ -7,15 +8,16 @@ namespace eden
       eosio::check(migration_sing.get().index() == new_value.index(),
                    "Cannot change current migration");
       migration_sing.set(new_value, contract);
+      push_event(migration_event{static_cast<uint16_t>(new_value.index())}, contract);
```

Comment:
This is the wrong place to emit the event.  This function is not used by the `migrate` action; it's used by other actions that affect the state of the migration.

Reply:
Could you point me to the correct place to emit the event please? What about the next one, inside `init()`?

Reply:
`init` happens at `genesis`. `migrate_some`.

Reply:
It should be emitted any time the type held in the variant changes.  That means `init` and `migrate_some`.  `set` is not needed as it explicitly forbids changing the type held by the variant.

Reply:
> `init` happens at `genesis`. `migrate_some`.

emitting the event from `init` allows the microchain to know what version of the contract was deployed at genesis.